Use of quarkus-agroal with GCP CloudSQL

jtama opened the pull request:
This pull request contains code to ease the use of GCP CloudSQL as well as documentation to use it without any code amendments.

Commit:
Add the hibernate `globally-quoted-identifiers` parameter.

dcdh committed:
Support Hibernate ORM's map entity mode

loicmathieu committed:
feat: projection

jmartisk committed:
Unified way for extensions to register metrics

jmartisk committed:
Turn datasources' metrics-enabled into a build time property

jmartisk committed:
Expose metrics for Agroal extension

vsevel committed:
Add Support to Multiple Vault KV Paths - Fixes Issue 5638 #6174

Sanne commented on extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSourceRuntimeConfig.java:
```diff
@@ -130,4 +130,16 @@
+     * Query executed to validate a connection.
+     */
+    @ConfigItem
+    public Optional<String> cloudSqlInstance;
```
isn't this replicating the `validationQuerySql` property ?

jtama-op replied:
Nope, this property value will end up in the jdbc url and will be used by the `SocketFactory` to provide access to the CloudSQL instance.

Sanne commented on extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSourceRuntimeConfig.java:
```diff
@@ -130,4 +130,16 @@
+     * When enabled jdbc url will be forged to enable google socket factory
+     */
+    @ConfigItem(defaultValue = "false")
+    public boolean useGcp;
```
Just wondering, would it be possible to automate this in any way?

Would be great if we could detect it's being run on GCP and not need such a configuration flag

jtama-op replied:
we can detect we are running on GCP by connecting to the `metadata.google.internal` as followed : 
```java 
try {
        Socket socket = new Socket("meta.google.internal", 80);
        System.out.println("Running in GCP!!");
    } catch (IOException e) {
        e.printStackTrace();
    }
```

But as stated in the [Deploying to google cloud documentation](docs/src/main/asciidoc/deploying-to-google-cloud.adoc), connecting with service account is just an option. I don't think Quarkus should force it. We could combine this detection and the presence of the cloudSqlInstance property to automate this.

Again as stated in the https://github.com/quarkusio/quarkus/issues/5824, for it to work as is, we would need to add a dependency to `com.google.cloud.sql:postgres-socket-factory:1.0.15` and/or `com.google.cloud.sql:mysql-socket-factory-connector-j-8:1.0.15` plus dependencies to the corresponding drivers.

As of now it is not possible as maven-enforcer-plugin will fail the build.

Alternatively, we could delegate the responsibility to the applications using the quarkus-agroal extension.

jtama-op replied:
I have removed postgres and postgres-socket-factory so that the build doesn't fail.

Again we could force the use of service account, but it would be making decisions instead of letting developper choose their connection mode

Sanne commented:
hey, many thanks for proposing this. It's exciting to see more support for CGP!

I hope you could help me figure out if we can make it really easy and slick to run on GCP, so without needing any additional configuration flags? Please see my review comments for details.

It would also be great as to not have configuration properties which only apply to a single cloud.

Sanne commented on docs/src/main/asciidoc/deploying-to-google-cloud.adoc:
```diff
@@ -0,0 +1,109 @@
+For this guide you need:
+
+* roughly 1 hour
+* having access to an Google cloud plateform project with owner rights.
```
a typo : s/plateform/platform/

jtama-op replied:
Done

jtama-op commented:
Links to #5824

gastaldi committed:
Add README.md for the Gradle plugin

Fixes #5804

Co-Authored-By: Guillaume Smet <guillaume.smet@gmail.com>

jtama-op commented:
Hi,
@Sanne I am not sure about the merge request process, do I need to do something more ?

manovotn committed:
Provided ManagedExecutor and ThreadContext beans should be default so that users can override them easily.

gsmet committed:
Move Elytron Security substitutions to their own module

gsmet committed:
Add the jsonp extension as it is required for OAuth2 authentication

geoand committed:
Introduce handling of more advanced generics cases

Co-authored-by: Stéphane Épardaud <stef@epardaud.fr>

mkouba committed:
Mailer - support Qute templates

- resolves #6158
- also collect all possible template variants and fix #6159

geoand committed:
Make JandexUtilTest more readable

Co-authored-by: Stéphane Épardaud <stef@epardaud.fr>

loicmathieu committed:
Fix a small documentation error

dmlloyd committed:
Let the delegate handlers decide to autoflush

mkouba committed:
Qute - escape expressions in html by default

- resolves #6155
- API changes:
 - introduce TemplateLocator
 - add Template#getVariant()

gwenneg committed:
Initialize com.microsoft.sqlserver.jdbc.KerbAuthentication at run time

gastaldi committed:
Add quarkus-kafka-streams-deployment to bom/deployment

gsmet committed:
Raise a few native tests timeout

aloubyansky committed:
Removed Quarkus bootstrap dependencies not used (directly) by the bootstrap core module

stuartwdouglas committed:
Improve error message if config properties conflict

gastaldi committed:
Move Launching JVM log to debug level

Fixes #6183

geoand committed:
Fix erroneous Kubernetes file name in docs

Fixes: #6187

FroMage committed:
Fix #5885: generics break panache repo enhancer

Do not enhance generic/abstract repos
Use proper generics resolution when looking up the repository type
arguments

FroMage committed:
Moved skipRepository up

FroMage committed:
Test for mongo version of #5885 bug

gsmet committed:
Use the bean index to scan subresources, check parameters and look for @Context

In the case of generated resource beans such as in Kogito, this was
causing failures.

It should fix the quickstart issue we have right now.

yrodiere committed:
Upgrade to Hibernate Search 6.0.0.Beta3

yrodiere committed:
Filter out the new "version" log from Hibernate Search on startup

aguibert committed:
Transfer JAX-B annotations from fields to accessor

For Panache entity enhancement we add @XmlTransient to the original fields
to force Jackson to use the generated accessor methods. However, this can
cause conflicts if the field was originally annotated with JAX-B annotations
that are mutually exclusive with @XmlTransient.

Signed-off-by: Andrew Guibert <andy.guibert@gmail.com>

dependabot-preview[bot] committed:
Bump flyway-core from 6.1.1 to 6.1.2

Bumps [flyway-core](https://github.com/flyway/flyway) from 6.1.1 to 6.1.2.
- [Release notes](https://github.com/flyway/flyway/releases)
- [Commits](https://github.com/flyway/flyway/compare/flyway-6.1.1...flyway-6.1.2)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

geoand committed:
Move Spring Data JPA classes to panache to avoid dev-mode CL issue

Fixes #6214

pedroigor committed:
[fixes #5926] - Attempt to make code flow test more stable

stuartwdouglas committed:
Run standalone RESTeasy in main thread pool

This means both Servlet and standalone will both
use the main thread pool, and it will avoid Vert.x
blocked thread warnings. These warnings are useless
as blocking IO can potentially block a thread for
a long time if request or response is large and the
connection is slow.

aguibert committed:
Support annotations with nested annotation arrays

stuartwdouglas committed:
Allow multiple headers with same name

Manually adding multiple JAX-RS Set-Cookie headers
would result in only a single one being sent to the client.

pedroigor committed:
cookie test

gsmet committed:
Add a quick workaround to get javadoc generation happy

mkouba committed:
ArC - activate request context for any observer notification

- resolves #6221

yrodiere committed:
Add more assertions for mass indexing in Hibernate Search

geoand committed:
Add a dev-mode test for the Spring Data JPA module

This probably should have been done as part of #6216
but time was of the essence for that one

yrodiere committed:
Work around a problem with Version.logVersion in Hibernate Search 6.0.0.Beta3

gsmet committed:
Some housekeeping on our pom files

paschi committed:
Fix file system path for overriding application.properties when using dev-mode with Gradle

gwenneg committed:
Include sun.security.util.Resources bundle in native image

stuartwdouglas committed:
Agroal dev mode test

gastaldi committed:
Minor cleanup in Gradle plugin

Using constants and avoiding casts

stuartwdouglas committed:
Add websockets dev mode test

geoand committed:
Add a dev-mode test for the Spring Security module

geoand committed:
Add a dev-mode test for the Spring Web module

jmartisk committed:
Devmode test for Metrics

mkouba committed:
Update mailer docs - mention MailTemplate

n1hility committed:
remove whitespace

remove whitespace

n1hility committed:
Workaround CI failure

Workaround CI failure

n1hility committed:
Workaround CI

Workaround CI

Commit:
Spring Boot properties extension

n1hility committed:
Update workaround

Update workaround

n1hility committed:
Update native-build-steps.yaml

viniciusfcf committed:
auth-mechanism property does not exist

auth-mechanism property does not exist

n1hility committed:
Complete work-around for CI (#6256)

Complete work-around for CI

geoand committed:
Fix broken call-out

dependabot-preview[bot] committed:
Bump debezium-core from 0.10.0.Final to 1.0.0.Final

Bumps [debezium-core](https://github.com/debezium/debezium) from 0.10.0.Final to 1.0.0.Final.
- [Release notes](https://github.com/debezium/debezium/releases)
- [Changelog](https://github.com/debezium/debezium/blob/master/CHANGELOG.md)
- [Commits](https://github.com/debezium/debezium/compare/v0.10.0.Final...v1.0.0.Final)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

geoand committed:
Make application version and name available as runtime properties

Fixes: #6255

Co-authored-by: Guillaume Smet <guillaume.smet@gmail.com>

mkouba committed:
Reactive routes - detect unordered conflicting routes

- resolves #6165

pavolloffay committed:
Put trace context into MDC 🎅🎄🎁

Signed-off-by: Pavol Loffay <ploffay@redhat.com>

ia3andy committed:
Fix table design on safari and mobiles

ia3andy committed:
Use uppercase only in placeholder for the config search

mejlholm committed:
Mention possibility to disable JSON for non prod

Remove <build> surrounding extension - it's not valid.
Add small example of how to use human readable logging for dev/test and structured JSON for production.

jaikiran committed:
issue-5942 Introduce a deterministic ordering, between extensions, for database schema updates

jaikiran committed:
Prevent UnsupportedOperationException in addSourcePaths

pivovarit committed:
Update writing-extensions.adoc

geoand committed:
Add tests for application name and version handling

jmartisk committed:
Prevent throwing NPE when recorded objects have null collections

michael-simons committed:
Fix package name of Neo4j health check.

geoand committed:
Bump RESTEasy to 4.4.2.Final

ia3andy committed:
Add note about Sentry re-licensing to BSL

geoand committed:
Ensure that NativeIT is only run during when using native profile

The other tests are only run when this profile is not active in order
to avoid running them multiple times (since they are slow as well)

Co-authored-by: Guillaume Smet <guillaume.smet@gmail.com>

loicmathieu committed:
Implements @ProjectionFor

michalszynkiewicz committed:
SmallRye FT 4.0.0

jaikiran committed:
issue-6280 Address Class-Path parsing issues with Java 11 and Windows, for tests launched using QuarkusDevModeTest

dependabot-preview[bot] committed:
Bump flyway-core from 6.1.2 to 6.1.3

Bumps [flyway-core](https://github.com/flyway/flyway) from 6.1.2 to 6.1.3.
- [Release notes](https://github.com/flyway/flyway/releases)
- [Commits](https://github.com/flyway/flyway/compare/flyway-6.1.2...flyway-6.1.3)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

maxandersen committed:
sentry.io is not opensource

gsmet committed:
Only have one warning about the non-open source Sentry license

Commit:
feat(flyway): integrate flyway's validate functionality into quarkus

geoand committed:
Use same all-open configuration for generated Kotlin Maven projects as Gradle

geoand committed:
Ensure that the Kotlin compiler adds method param names to bytecode

We already do this for Java, so let's make it consistent for Kotlin too

Fixes: #6041

gastaldi committed:
Added back methods mistakenly considered unused in QuarkusPluginExtension

These methods were removed in 3265edd082955d6f5b0498dfeb79033c2c073219 because they were considered unused but they are used in the Kotlin Gradle build.
Added tests for all SourceTypes now

Fixes #6311

gastaldi committed:
Show stacktrace when building the gradle plugin

gastaldi committed:
Disabling Scala gradle build test for now

gastaldi committed:
Adopt new label nomenclature in issue templates

Mihai-B commented:
After this is merged, in what version will it be available ?

gastaldi committed:
Fixes Illegal char error in Gradle plugin

gastaldi committed:
Revert "Disabling Scala gradle build test for now"

This reverts commit b3ea880dabcbbeb135f5d47879e91838aa11016a.

gastaldi committed:
Set className and path to trigger sample code generation

gastaldi committed:
basic-rest/scala/build.gradle-template.ftl requires scala-library

Fixes #6316

geoand committed:
Increase timeout in one of the native image stages that frequently fails in CI

geoand committed:
Ensure that Kotlin Data classes with default values work in native with JAX-RS

Fixes: #3954

gsmet committed:
Add missing descriptions to recently added extensions

gsmet committed:
Tweak the Sentry extension description

gsmet committed:
Fix a source extract in Qute documentation

gsmet committed:
Be a bit more consistent in Qute documentation indentation

gsmet committed:
Fix a minor indentation issue in Gradle tooling documentation

gsmet committed:
Review the Logging to Sentry documentation

lordofthejars committed:
Fixes spaces to be double spaces in all the examples

Commit:
Add SSL port in server start log message

rquinio committed:
Handle javax.inject.Named on implict singleton bean in spring-di

kdnakt committed:
fix typo

gastaldi committed:
Using constants in Gradle tests

loicmathieu committed:
Fix Elytron properties perf issue

loicmathieu committed:
Fix Elytron JDBC perf issue

loicmathieu committed:
Provides additional configuration options for logging-gelf

gastaldi committed:
Support physicalNamingStrategy

Fixes #6346

ianwormsbecker-ssimwave committed:
Update docs with import for RegisterForReflection

machi1990 committed:
fix: DynamicProxySupport is not thread safe and Neo4j doesn't work in native mode

The issues are fixed by updating GraalVM version to 19.3.0.2

Fixes https://github.com/quarkusio/quarkus/issues/6115

machi1990 committed:
ci(neo4j): add neo4j tests in CI

machi1990 committed:
docs(qute): add link to Qute reference guide in the Qute templating engine guide

Fixes https://github.com/quarkusio/quarkus/issues/6367

kdnakt committed:
docs: fix project artifact id for jackson

kdnakt committed:
fix project name in the link to legumes.html

geoand committed:
Ensure that MySQL integration test works properly with Docker Maven plugin

Fixes: #6372

gsmet committed:
Upgrade Jackson to 2.10.1

gsmet committed:
Remove substitutions incorporated into GraalVM 19.3.0.2

pedroigor committed:
[fixes #4448] - Support OidcTenantConfig resolution

loicmathieu committed:
Add Emitter usage to reactive messaging guides

geoand committed:
Properly initialize Maven profile options during bootstrap

Fixes: #6383

devauxbr committed:
enable hot reload on application.yaml

geoand committed:
Create BuildTimeRunTimeDefaultValuesConfigSource only once

geoand committed:
Document how to retrieve the active profile

loicmathieu committed:
Add JSON-B and Jackson serialization to the Kafka guide

loicmathieu committed:
Configure additional fields with a Map

irenakezic committed:
Add Hibernate Validator dev mode tests

nmcl committed:
Improved STM text

Will make further changes once I've modified the quickstart code itself.

https://github.com/quarkusio/quarkusio.github.io/issues/411

https://github.com/quarkusio/quarkusio.github.io/pull/417

devauxbr committed:
add application.yaml hot reload automated test

devauxbr committed:
add the ability to include full MDC in logging-gelf extension configuration

gunnarmorling committed:
Typo fix in Neo4j guide

gsmet committed:
Upgrade Jackson to 2.10.2

gsmet committed:
Take into account the web.xml MIME mappings in the Undertow extension

Fixes #6399

michael-simons committed:
Improve the Neo4j guide.

The Neo4j driver has been updated to 4.0.0 final and Neo4j 4.0 can be
used from Docker, the preview flag can now be removed.

The final driver revealed some errors in the existing guide:

The reading of the transactional result of a transactional function
needs to happen inside the same transaction.
Therefor the guide needs a fix so that the composition of the future
extracting the node happens in the transaction, not outside.

rsvoboda committed:
Typo fixes

Commit:
Add possibility to override the logging configuration of each handler on category level  A

geoand committed:
Fix Kubernetes manifest port handling

This is needed because of the config changes that went in 1.1.
One thing that needed to be removed because of those changes was
the warning we used to produce about potential mismatch between
the port in the Kubernetes manifests and the port set at runtime

Fixes: #6408

iocanel committed:
doc (#6303): Fix table structure and alignment.

ia3andy committed:
Make quarkus.log.sentry.dsn required only when sentry is enabled

mkouba committed:
ArC - optimize CreationalContext

- do not always store all dependent instances in CC
- also introduce io.quarkus.arc.InjectableInstance
- resolves #5281

dependabot-preview[bot] committed:
Bump flyway-core from 6.1.3 to 6.1.4

Bumps [flyway-core](https://github.com/flyway/flyway) from 6.1.3 to 6.1.4.
- [Release notes](https://github.com/flyway/flyway/releases)
- [Commits](https://github.com/flyway/flyway/compare/flyway-6.1.3...flyway-6.1.4)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

gsmet committed:
Check that there is a pluginOptions child before iterating on it

Fixes #6428

viniciusfcf committed:
Fix Logger.getLogger example parameter

mkouba committed:
Qute - more meaningful exception for missing parameter declaration class

- resolves #6366

mkouba committed:
Qute - support @TemplateExtension declared on a class

- also update docs and add some tests
- resolves #6438

manovotn committed:
Extending smoke test for SR context propagation integration.

mkouba committed:
InstanceHandle#destroy - throw ContextNotActiveException if appropriate

manovotn committed:
Arc should only emit once log msg per missing class in index.

kdnakt committed:
docs: fix link

rsvoboda committed:
Docs typo fixes

dependabot-preview[bot] committed:
Bump mariadb-java-client from 2.5.2 to 2.5.3

Bumps [mariadb-java-client](https://github.com/mariadb-corporation/mariadb-connector-j) from 2.5.2 to 2.5.3.
- [Release notes](https://github.com/mariadb-corporation/mariadb-connector-j/releases)
- [Changelog](https://github.com/mariadb-corporation/mariadb-connector-j/blob/master/CHANGELOG.md)
- [Commits](https://github.com/mariadb-corporation/mariadb-connector-j/compare/2.5.2...2.5.3)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

gwenneg committed:
Remove unused Vert.x GraalVM substitution

gastaldi committed:
Add tests for PhysicalNamingStrategy feature

gastaldi committed:
Use StandardCharsets.UTF_8 instead of UTF-8 String

stuartwdouglas committed:
Allow JAX-RS mapping in web.xml

Fixes #6426

tobad357 committed:
This fixes issue #6442 that project dependencies end up twice on the quarkusDev classpath

gsmet committed:
Remove useless version from the reactive * client poms

gastaldi committed:
Properties should not be reused between tests

haraldatbmw committed:
Make native executable executable in Dockerfile

mkouba committed:
Qute parser - improve error messages

- resolves #6164

dependabot-preview[bot] committed:
Bump kubernetes-client-bom from 4.6.4 to 4.7.0

Bumps [kubernetes-client-bom](https://github.com/fabric8io/kubernetes-client) from 4.6.4 to 4.7.0.
- [Release notes](https://github.com/fabric8io/kubernetes-client/releases)
- [Changelog](https://github.com/fabric8io/kubernetes-client/blob/master/CHANGELOG.md)
- [Commits](https://github.com/fabric8io/kubernetes-client/compare/v4.6.4...v4.7.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

gwenneg committed:
Fix GraalVM SDK version in documentation

geoand committed:
Add support for kebab case in @ConfigProperties

Fixes: #6457

loicmathieu committed:
Use mongodb-driver-sync instead of mongo-java-driver

gsmet committed:
Fix a copy/paste error in writing-extensions.adoc

This is a detail but it's breaking something I'm working on.

gsmet committed:
Generate PDFs for the documentation and build an assemble zip

The idea is to have a link on the website allowing to download all the
PDF documents at once.

First step for fixing
https://github.com/quarkusio/quarkusio.github.io/issues/419 .

mattyb678 committed:
Add ability to disable Swagger UI

mattyb678 committed:
Fix spelling in extension doc

geoand committed:
Properly show configuration errors in the output

Fixes: #6466

RotBolt committed:
Update getting-started.adoc for windows

Edit commands for cmd & powershell on windows

geoand committed:
Make kebab case the default in @ConfigProperties

gastaldi committed:
graalVmHome should be optional in gradle buildNative

gastaldi committed:
Do not assume non-existent source dirs as working and source dir

Fixes #6485

geoand committed:
Update core/runtime/src/main/java/io/quarkus/runtime/configuration/ConfigDiagnostic.java

Co-Authored-By: David M. Lloyd <david.lloyd@redhat.com>

marceloverdijk committed:
Updated Mockito version

Mockito version defined in bom was quite old already, consider using latest Mockito version.

marceloverdijk committed:
Updated Mockito version

geoand committed:
Ensure that Java 12+ interfaces of java.lang.String are as CDI types

This is needed because when building with Java 12+ but targeting Java 8-11,
we need to ensure that the extra interfaces that java.lang.String implements
do not end up in the generated bytecode.

Fixes: #5062

marceloverdijk committed:
Added Mockito to .dependabot config

viniciusfcf committed:
Using TimeUnit.NANOSECONDS instead of NANOSECONDS

Making code example clearer

gwenneg committed:
Enable SSL native support for quarkus-jsch

gsmet committed:
Set higher timeouts for JVM CI jobs

gastaldi committed:
Update Mockito in tools project

gastaldi committed:
Use same JUnit version and artifact used in quarkus BOM

cescoffier committed:
Remove non-used Docker files

cescoffier committed:
Use UBI in the JVM variant of the Dockerfile

I added a note (as a comment) about -XshowSettings:vm. So the user can enable it easily and check the amount of heap used by the application.

mkouba committed:
Qute "if" section - support multiple conditions

loicmathieu committed:
Update to logstash-gelf 1.14.0 and remove the substitution

Fixes #6434

cescoffier committed:
update and align RX Java 2 version to 2.2.17

cescoffier committed:
Explicitly set the UBI version to 8.1 in the Dockerfile used for the native executable.

dependabot-preview[bot] committed:
Bump mysql-connector-java from 8.0.18 to 8.0.19

Bumps [mysql-connector-java](https://github.com/mysql/mysql-connector-j) from 8.0.18 to 8.0.19.
- [Release notes](https://github.com/mysql/mysql-connector-j/releases)
- [Changelog](https://github.com/mysql/mysql-connector-j/blob/release/8.0/CHANGES)
- [Commits](https://github.com/mysql/mysql-connector-j/compare/8.0.18...8.0.19)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

dependabot-preview[bot] committed:
Bump httpcore from 4.4.12 to 4.4.13

Bumps httpcore from 4.4.12 to 4.4.13.

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

Commit:
added elytron ldap authentication

cescoffier committed:
Add RX Java 2 to the list of dependency maintained by Dependabot

Note that RX Java 3 is using a different groupId and so is not handled by this PR. There is no plan right now to switch to RX Java 3.

mkouba committed:
Swallow exceptions thrown from Bean.destroy()

- otherwise an error prevents proper container shutdown
- resolves #6529
- also fix EventImpl - use the correct event type

gastaldi committed:
Support ImplicitNamingStrategy

Fixes #6546

Commit:
Improving the mailer adoc by adding quarkus.mailer.mock=false to sendgrid example

The sendgrid example in the current state will not work because the quarkus.mailer.mock=false is not present. If you just try to send the email it will fail and you will have to read the whole testing section to find out that the quarkus.mailer.mock is true by default.

I find this miss leading I personally have lost around 40mins finding and debugging why the email is not send and finding that I am in a mock mode.

I greatly believe that this will help the others using mailer to send email with or without sendgrid.

Maybe it is wise to even add a comment somewhere bellow the properties similar on how the password vault is explained.

geoand committed:
Restore the ability to use a system property for remote-dev

Fixes: #6550

manovotn committed:
Introduce transactional observer support.

gastaldi committed:
ObjectStoreEnvironmentBean.getObjectStore should not cache the user.home during build-time

Fixes #6553

kdnakt committed:
doc: use public inner class for reflection

geoand committed:
Make sure we read properties with UTF8 encoding during live-reload

aguibert committed:
Update Yasson to v1.0.6

hutchig committed:
Some tiny English fixes from reading guides

Signed-off-by: Gordon Hutchison <Gordon.Hutchison@gmail.com>

aguibert committed:
Add 'org.eclipse:yasson' to dependabot config

rsvoboda committed:
Fix spelling in docs

Sanne committed:
Disable automatic association management in Hibernate ORM entities

mkouba committed:
Qute "if" section - update docs

Sanne committed:
Fix the Panache / Hibernate ORM tests to not rely on automatic association management

gwenneg committed:
Support both GraalVM 19.2.1 and 19.3.1

sberyozkin committed:
Update to smallrye-jwt 2.0.12

gwenneg committed:
Fix quarkus-integration-test-infinispan-embedded in native mode

gwenneg committed:
Disable quarkus-integration-test-jsch in native mode temporarily

gwenneg committed:
Remove useless project.version in quarkus-narayana-stm-deployment

mkouba committed:
Qute value resolver generator - always skip static methods

- resolves #6396
- also add TemplateData#ignoreSubclasses()

iocanel committed:
fix (#6562): Openshift manifests now reference the correct artifact.

ia3andy committed:
Fix GELF guide link

Closes #6577

iocanel committed:
fix (#6580) Add openshift, knative and more to the list of supported
prefixes in the kubernetes extension.

iocanel committed:
test (#6580): Add integration-test to verify the properties with the
openshift prefix are well respected.

michalszynkiewicz committed:
SmallRye FT 4.0.0: added service provider for CP executor factory

gwenneg committed:
Disable JNI by default with GraalVM 19.2.1

loicmathieu committed:
Manage Instant as MongoDB ISODate

sberyozkin committed:
Support custom principal claims

geoand committed:
Disable failing OpenTracing TCK module

See: https://github.com/eclipse/microprofile-opentracing/issues/165
for more details

mkouba committed:
Make use of QuteConfig.suffixes when validating template injections

- also change the default value to support VS Code extension

rsvoboda committed:
Relax output checks to pass on Windows with cmder

gwenneg committed:
Fix H2 native DeletedElementException

stuartwdouglas committed:
Fix build hang on systems with low disk space

stuartwdouglas committed:
RESTeasy standalone does not respect max request size

Fixes #6590

dmlloyd committed:
Add Netty substitution that addresses Camel Quarkus issue

geoand committed:
Update extensions/netty/runtime/src/main/java/io/quarkus/netty/runtime/graal/NettySubstitutions.java

Co-Authored-By: Guillaume Smet <guillaume.smet@gmail.com>

mkouba committed:
Tx observers - minor cleanup plus reduce the number of lamdas

Mihai-B commented on extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/AbstractDataSourceProducer.java:
```diff
@@ -195,6 +196,21 @@ public boolean isValid(Connection connection) {
+                throw new RuntimeException("Cloud sql instance should match the pattern project-id:zone:cloudSqlInstance");
+            }
+            agroalConnectionFactoryConfigurationSupplier.jdbcUrl(
+                    MessageFormat.format("{0}?socketFactory=com.google.cloud.sql.postgres.SocketFactory&cloudSqlInstance={1}"
```
The socketFactory is bound to postgres driver and if you would like to use a different driver is not possible with current implementation. Is it possible to make this configurable from a property as well ?

jtama-op replied:
Currently there are only two supported SGBD postgres and mysql, but yes I should make this configurable.

Mihai-B replied:
But I think @Sanne is correct, this implementation is very GCP integrated. maybe we should switch a little bit the way we make the implementation for this.

I think the easiest way is to use the `agroalConnectionFactoryConfigurationSupplier.jdbcProperty(<property_name>, <value>)` method

jtama-op replied:
I tried this but it didn't worked. I will re-take a loot at it. 

In fact I think I should create a separate extension for gcp sql datasources.

Mihai-B replied:
Can you check: https://github.com/quarkusio/quarkus/pull/6634 ?

jtama-op replied:
I think it won't work (i tried this already)

gsmet committed:
Handle association manually as automatic association management was disabled

Update the Hibernate Search Elasticsearch quickstart.

See https://github.com/quarkusio/quarkus/pull/6569

Mihai-B commented on extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/DataSourceRuntimeConfig.java:
```diff
@@ -130,4 +130,16 @@
+    public boolean useGcp;
+
+    /**
+     * Query executed to validate a connection.
```
I think the message here is wrong

jtama-op replied:
You are right I will change this

Mihai-B commented on extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/AbstractDataSourceProducer.java:
```diff
@@ -195,6 +196,20 @@ public boolean isValid(Connection connection) {
+                throw new RuntimeException("Cloud sql instance should match the pattern project-id:zone:cloudSqlInstance");
+            }
+            agroalConnectionFactoryConfigurationSupplier.jdbcUrl(
+                    MessageFormat.format("{0}?socketFactory=com.google.cloud.sql.postgres.SocketFactory&cloudSqlInstance={1}",
```
Maybe we should use the `agroalConnectionFactoryConfigurationSupplier.jdbcProperty(<property_name>, <value>) ` method here ?

jtama-op replied:
I don't think it works

Mihai-B reviewed (commented):
just added some ideas

mkouba committed:
Fix simple scheduler cron triggers in JDK9+

- resolves #6600

gwenneg committed:
Forbid native image generation from GraalVM 19.3.0.2

machi1990 committed:
docs: update precedence note between ".properties" and ".yaml" config files

The .yaml file (254 ordinal -
https://github.com/quarkusio/quarkus/blob/5db3f0f090b00821db3edf5e30759e3446c20712/extensions/config-yaml/runtime/src/main/java/io/quarkus/config/yaml/runtime/ApplicationYamlProvider.java#L36
) takes precedence over a .properties (250 ordinal -
https://github.com/quarkusio/quarkus/blob/a91a034b855cae3f8d474951cb33e8d05707dc13/core/runtime/src/main/java/io/quarkus/runtime/configuration/ApplicationPropertiesConfigSource.java#L52)
files

machi1990 committed:
docs: provide a shell command that adds a config yaml extension in the config guide

lordofthejars committed:
Fix multidatasource example

sberyozkin committed:
Disable OIDC principal claim test

gastaldi committed:
Enable JNI in elytron-security-ldap

gastaldi committed:
Using simplified classes

Fixes #6606

mkouba committed:
Update CDI ref guide

geoand committed:
Trim graalvm additional build arguments

Fixes: #6598

gastaldi committed:
Add elytron-security-ldap to CI pipeline

gastaldi committed:
Disabling native image tests because of the following error

Fatal error: Failed to leave the current IsolateThread context and to detach the current thread. (code 12)

kdnakt committed:
doc: fix compilation error

kdnakt committed:
add missing import of RequestScoped annotation

jtama committed:
Corrects javadoc

gsmet commented:
This PR is invalid and contains a lot of unrelated commits.

Closing. Please open a new PR to pursue this.
